controllers/openshift_controller.go

```diff
@@ -1805,6 +1805,55 @@ func (r *KataConfigOpenShiftReconciler) processDoneNode(node *corev1.Node) error
 	return nil
 }
 
+// If multiple errors occur during execution of this function the last one
+// will be returned.
+func (r *KataConfigOpenShiftReconciler) updateStatusNew() error {
+
+	if r.kataConfig.Status.UnInstallationStatus.InProgress.IsInProgress != corev1.ConditionTrue &&
+		r.kataConfig.Status.InstallationStatus.IsInProgress != corev1.ConditionTrue {
+		return nil
+	}
+
+	err, nodeList := r.getNodes()
+	if err != nil {
+		return err
+	}
+
+	r.clearInstallStatus()
+	r.clearUninstallStatus()
+
+	r.kataConfig.Status.TotalNodesCount = func() int {
+		err, nodes := r.getNodesWithLabels(r.getNodeSelectorAsMap())
+		if err != nil {
+			r.Log.Info("Error retrieving kata-oc labelled Nodes to count them", "err", err)
+			return 0
```

Comment:
Is this right ? Shouldn't `updateStatusNew()` bail out if it cannot retrieve the list of nodes ?

Reply:
It's a judgment call I guess.  This failure would only prevent setting the `TotalNodesCount` correctly but the function would still be able to do the bulk of its job (it has already retrieved `nodeList` above successfully otherwise we wouldn't be here in the first place).

Given this, and also the fact that this problem will likely go away next time around `updateStatus()` is called, I thought it'd be a shame to leave all of the status not updated and stale just because we couldn't count nodes, probably temporarily.

Reply:
Fair enough.